### DISCUSSION TITLE
Acoustics in moving liquid media

### DISCUSSION THREAD

**Discussion by Ambr0K** [SPECFEM/specfem3d#1765] [SPECFEM3D General]
Dear all.

We're currently thinking about the introduction of staggered ultrasonic sensor arrays at the edge of a cylindrical tank of water in a pool type research reactor to (hopefully) obtain a 3D field of both the temperature, as well as the velocity and the direction of the flow of the coolant water. We've looked into the popular codes like k-wave, but they lack the implementation of fluid velocity terms (as outlined in Eq. 2.84 amd 2.86 @ https://doi.org/10.1201/b18922). The easy way would be to just depend on the Time of flight (TOF) measurements, but the resolution of accuracy would be somewhat limiting. Right now, the most feasible option seems to be using TOF for velocity and wave propagation for temperature tomography.

My question really is, whether the governing acoustic equations include the option for a moving fluid medium, which we could use to asses the responses of individual sensors in the array, and to use this data for training for the real implementation in an experiment.

Thank you in advance!

**Comment by danielpeter** [comment]
SPECFEM implements the linearized equations of motions for a medium at rest. so, I'm afraid this code won't help much as there is no option for a moving fluid medium at the moment. 

if you feel brave and want to contribute this to this open-source package , that would great!  to add such a code capability, you would probably add an SEM implementation of the additional advection term $2({\bf v}_0 \cdot {\bf \nabla} ) \frac{\partial}{\partial_t}$  that arises due to an initial velocity field ${\bf v}_0$ for the acoustic wave equation.
